### PR TITLE
Add C FFI for LocaleCanonicalizer

### PR DESCRIPTION
**Pull request by dminor**
Fix for #756.

### CODE REVIEW

**Review by zbraniecki — DISMISSED** [review]
`comment (nb)`: reading this I realized that we hide the reasons constructor fails in C API. I hope we'll improve it one day since constructor failures are likely going to be most common in production environments.

File: ffi/capi/examples/locale/test.c [hunk 1]
```diff
@@ -3,23 +3,71 @@
...
+    // Test minimize.
+    write = icu4x_simple_writeable(output, 40);
+    locale = icu4x_locale_create("zh-Hant", 7);
+    icu4x_localecanonicalizer_minimize(lc, locale);
+    success = icu4x_locale_tostring(locale, &write);
+    if (!success) {
+        return 1;
+    }
+    // expect "zh-TW"
+    printf("Output is %s\n", output);
```
Comment (Manishearth) [inline ffi/capi/examples/locale/test.c:55]: issue: we should probably actually test that this matches expectations

(make sure the locale test is being run in Makefile.toml, btw, it's not at the moment)

**Review by Manishearth — DISMISSED** [review]
looks pretty straightforward, glad it worked out to be mostly along the lines of the existing FFI!

**Comment by sffc** [comment]
> `comment (nb)`: reading this I realized that we hide the reasons constructor fails in C API. I hope we'll improve it one day since constructor failures are likely going to be most common in production environments.

Tracking issue, last discussed 2021-05-20 and assigned to @Manishearth for 0.4: https://github.com/unicode-org/icu4x/issues/703

**Review by Manishearth — DISMISSED** [review]

**Review by Manishearth — APPROVED** [review]

**Pull request merged**